Code/Framework/AzFramework/AzFramework/Input/Devices/Keyboard/InputDeviceKeyboard.h

```diff
@@ -370,9 +370,20 @@ namespace AzFramework
         // Reflection
         static void Reflect(AZ::ReflectContext* context);
 
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        // Foward declare the internal Implementation class so it can be passed into the constructor
+        class Implementation;
+
+        ////////////////////////////////////////////////////////////////////////////////////////////
+        //! Alias for the function type used to create a custom implementation for this input device
+        using ImplementationFactory = typename Implementation*(*)(InputDeviceKeyboard&);
```

Comment:
question: Also is it worth just making this an `AZStd::function` to make this a little more flexible as opposed to a pure function pointer? Not a huge deal but in this case the overhead is small and increased flexibility could be useful

Reply:
Got you makes sense, thanks for clarifying :+1: